incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -853,6 +861,38 @@ private int decodeConnect(
                 server.onDecodeError(traceId, authorization, reasonCode);
                 server.decoder = decodeIgnoreAll;
             }
+            else if (progress == limit && !session)
+            {
+                if (isSetPassword(flags) || isSetUsername(flags) || isSetWillFlag(flags))
+                {
+                    reasonCode = MALFORMED_PACKET;
+                }
+                server.doEncodeConnack(traceId, authorization, reasonCode, server.assignedClientId,
+                    false, null);
+                server.connected = true;
+                server.decoder = decodePacketType;
+            }
```

Comment:
This `else-if` block doesn't seem right, because the logic of next decoder state is no longer managed by the decoder, it is ignored here, and set in onSessionWindow.

I think it would be better to keep the decoder assignment consistently in the decoders, not in the server.onXX methods, and naturally trigger decodeNetwork from onSessionWindow in case anything is pending such as will message.

Reply:
This decoder assignment here is needed when session support is turned off for the simplicity of the tests, so we can't receive sessionWindow.

Reply:
onDecodeConnectPayload should recognise that the remainingLength is 0, and "skip" decoding, and onDecodeConnectPayload should set the decodePacketByte